common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -1607,7 +1622,8 @@ def cache_and_return(parent_loc):  # pylint:disable=missing-docstring
                 all_parents.append(parent)
 
             # since we sorted by SORT_REVISION_FAVOR_DRAFT, the 0'th parent is the one we want
-            if published_parents > 1:
+            # If multiple draft parents found, select the non-orphaned
+            if published_parents > 1 or all_parents > 1:
```

Comment:
See comment above:
```python
# there could be 2 different parents if
#   (1) the draft item was moved or
#   (2) the parent itself has 2 versions: DRAFT and PUBLISHED
#  if there are multiple parents with version PUBLISHED then choose from non-orphan parents
```
We shouldn't be concerned about `all_parents` here, I don't think.
Also `all_parents` is a list, while `published_parents` is a number. Also, it looks like orphaned parents get filtered out in the next line.